src/flint/test/test.py

```diff
@@ -1584,6 +1584,167 @@ def test_pickling():
         obj2 = pickle.loads(s)
         assert obj == obj2
 
+def test_fmpz_mod():
+    from flint import fmpz_mod_ctx, fmpz
+    
+    # Context tests
+    F163 = fmpz_mod_ctx(163)
+    assert F163.modulus() == 163
+
+    big = 2**1024 - 1
+    Fbig = fmpz_mod_ctx(2**1024 - 1)
+    assert Fbig.modulus() == big
+
+    # Rich comparisons 
+    assert raises(lambda: F163(123) > 0, TypeError)
+    assert raises(lambda: F163(123) >= 0, TypeError)
+    assert raises(lambda: F163(123) < 0, TypeError)
+    assert raises(lambda: F163(123) <= 0, TypeError)
+
+    assert (F163(123) == 123) is True
+    assert (F163(123) == 123 + 163) is True
+    assert (F163(123) == 1) is False
+    assert (F163(123) != 1) is True
+
+    assert (F163(123) == F163(123)) is True
+    assert (F163(123) == F163(123 + 163)) is True
+    assert (F163(123) == F163(1)) is False
+    assert (F163(123) != F163(1)) is True
+
+    # Is one, zero, canoncial
+    assert (F163(0) == 0) is True
+    assert F163(0).is_zero() is True
+    assert not F163(0)
+    assert not F163(163)
+    assert F163(1).is_one() is True
+    assert F163(164).is_one() is True
+    assert F163(1).is_one() is True
+    assert F163(2).is_one() is False
+    assert F163(123).is_canonical() is True
```

Comment:
Can `is_canonical` ever be False?

This seems like the kind of thing that makes sense in C but where the type is actually `fmpz` but not in Python where we can wrap everything up in an `fmpz_mod` and ensure that `is_canonical` is always true.

Reply:
Yeah, I have a TODO about this. I think the proper use case could be in `__mul__`, as we can only multiply `a*b` when both `a,b` are canonical. 

For the case when `a` is an `fmpz_mod` but `b` is any integer, we use `fmpz_mod_mul` but we must first reduce `b  mod n`, I suppose one could only do this reduction if `fmpz_mod_is_canonical` returns false, but just always reducing seemed fine?

I think the method on the class should be removed.

Reply:
I think that `F163(anything)` should either raise an error or return a canonical `fmpz_mod` in `F163`. In other wose the `is_canonical` method for `fmpz_mod` should always return True (and so the method is not needed and should be removed to avoid confusion).